metadata-ingestion/src/datahub/ingestion/source/nifi.py

```diff
@@ -374,6 +377,22 @@ def __init__(self, config: NifiSourceConfig, ctx: PipelineContext) -> None:
                 logger.error("Failed to get token")
                 self.report.report_failure(self.config.site_url, "Failed to get token")
 
+            self.session.headers.update(
+                {
+                    "Authorization": "Bearer " + token_response.text,
+                    # "Accept": "application/json",
+                    "Content-Type": "application/json",
+                }
+            )
+        elif self.config.auth is NifiAuthType.KERBEROS:
+            token_response = self.session.post(
+                url=urljoin(self.config.site_url, KERBEROS_TOKEN_ENDPOINT),
+                auth=HTTPSPNEGOAuth(),
+            )
+            if not token_response.ok:
+                logger.error("Failed to get token")
+                self.report.report_failure(self.config.site_url, "Failed to get token")
```

Comment:
Should we error out here or something? Awkward because its' in the init function but doesn't seem like anything would work after this. Same goes for thesingle user auth above

Reply:
that's right, I just reused the old pattern here.

I'll probably move this token generation logic inside get_workunits and in case of non-ok response, report failure and then immediately return.